### ISSUE TITLE
False positive InvalidAttribute: `#[\Override]` on properties is valid in PHP 8.5

### ISSUE THREAD

**Issue by alies-dev** [vimeo/psalm#11890]
### Summary

On PHP 8.5, `#[\Override]` is allowed on **properties** ([RFC: Add `#[\Override]` to properties](https://wiki.php.net/rfc/marking_overriden_properties)). Psalm reports a false positive `InvalidAttribute` for that valid usage.

### Reproduction

```php
<?php

class A
{
    protected int $x = 1;
}

class B extends A
{
    #[\Override]
    protected int $x = 2;
}
```

### Expected

No error. PHP 8.5 accepts this:

```
$ php -r 'class A { protected int $x = 1; } class B extends A { #[\Override] protected int $x = 2; } echo "accepted\n";'
accepted
```

### Actual

```
ERROR: InvalidAttribute - Attribute \Override cannot be used on a property (see https://psalm.dev/242)
```

### Root cause

The bundled stub `stubs/CoreGenericAttributes.phpstub` declares `Override` with `TARGET_METHOD` only:

```php
/** @psalm-immutable */
#[Attribute(Attribute::TARGET_METHOD)]
final class Override
{
    public function __construct() {}
}
```

As of PHP 8.5 the real target set is `method | property`. Confirmed against the engine (class constants are **not** a valid target, so they must be excluded):

```
$ php -r 'class A { const X = 1; } class B extends A { #[\Override] const X = 2; }'
PHP Fatal error: Attribute "Override" cannot target class constant (allowed targets: method, property)
```

### Suggested fix

```diff
-#[Attribute(Attribute::TARGET_METHOD)]
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_PROPERTY)]
 final class Override
```

Ideally guarded so `TARGET_PROPERTY` is only treated as valid under PHP 8.5+ target analysis.

### Environment

- Psalm 7.0.0-beta19
- PHP 8.5.7
- Stub unchanged on `master` / `5.x` at time of filing.
